### PR TITLE
[stable/concourse] Add optional datadog metrics config

### PR DESCRIPTION
**Pull request by wendorf**
**What this PR does / why we need it**:
Concourse 3.13.0 adds support for exporting metrics to Datadog (see concourse/atc#269). This adds support for the new flags, and bumps the Concourse version.

### CODE REVIEW

File: stable/concourse/values.yaml
Comment (unguiculus) [inline stable/concourse/values.yaml]: Camel case, please.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by wendorf** [comment]
Thanks for the feedback, @unguiculus . I fixed the key names, and made a couple other changes:

* Values are used to define environment variables, which are then used to set command flags. This brings the config more in line with other parameters.
* Add `web.metrics.datadog.agentHostUseHostIP` to set agent host IP to the underlying node's IP

**Comment by unguiculus** [comment]
cc @frodenas @viglesiasce @william-tran @nexeck

**Review by unguiculus — APPROVED** [review]

**Comment by william-tran** [comment]
@wendorf if upgrading to 3.13, this also needs the changes in https://github.com/kubernetes/charts/pull/5153, would you be willing to incorporate those changes in to this PR?

**Comment by wendorf** [comment]
Certainly. I'm a bit confused about the need to do so, though. When I was
first working on it, I was expecting to need to open 7799, but saw that my
test deployment was operating fine, and also saw nothing in the logs
indicating a problem. We're now using this PR in our production instance
with no problems.

So it's kinda spooky. I don't mind opening the port, but I'd like to
understand why it's working fine for us if it's actually required. I don't
want to hold up the merging of this PR, so I can just add it, but I'd love
a second pair of eyes on this confusion.

**Comment by william-tran** [comment]
@wendorf yeah that certainly is weird. I'm gonna /ok-to-test for now and see what happens while I do some testing on my end. I'd really like to get to 3.13.0 as well.

**Comment by william-tran** [comment]
@wendorf So it looks like you don't need to specify a port for it to be reachable from within the network: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#container-v1-core
> `ports` List of ports to expose from the container. Exposing a port here gives the system additional information about the network connections a container uses, but is primarily informational. Not specifying a port here DOES NOT prevent that port from being exposed. Any port which is listening on the default "0.0.0.0" address inside a container will be accessible from the network.

**Comment by wendorf** [comment]
Okay, that doesn't match the internal model I had. Thanks for investigating! I'll add the port to the service for completeness/explicitness. Would you also like me to add the other flags in #5153 then, or would you prefer to have that PR rebased on this once it gets merged?

**Review by william-tran — APPROVED** [review]

**Comment by william-tran** [comment]
@wendorf I'm going to make some updates to #5153 which will add the rest. Thanks!

**Pull request merged**
